content/ko/docs/topics/charts_hooks.md

```diff
@@ -0,0 +1,205 @@
+| `post-install`   | 쿠버네티스에 모든 리소스가 로드된 후에 실행된다.                                                                   |
+| `pre-delete`     | 삭제 요청 중에서, 쿠버네티스에서 리소스가 삭제되기 전에 실행된다.                                                         |
+| `post-delete`    | 삭제 요청 중에서, 릴리스의 모든 리소스가 삭제된 후 실행된다.                                                            |
+| `pre-upgrade`    | 업그레이드 요청 중에서, 템플릿은 렌더링되었지만 리소스는 업데이트되기 전에 실행된다.                                            |
+| `post-upgrade`   | 업그레이드 요청 중에서, 모든 리소스가 업그레이드된 후 실행된다.                                                             |
+| `pre-rollback`   | 롤백 요청 중에서, 템플릿은 렌더링되었지만 아직 아무 리소스도 롤백되지 않은 시점에 실행된다.                                                    |
+| `post-rollback`  | 롤백 요청 중에서, 모든 리소스가 수정된 후에 실행된다.                                                                 |
+| `test`           | 헬름 test 하위 명령어가 호출될 때 실행된다. ([테스트 문서 보기](/docs/chart_tests/))                                  |
+
+_헬름 3에서는 `crd-install` 훅은 제거되었고 대신 `crds/` 디렉터리를 
+사용한다는 점을 잊지 말자._ 
+
+## 훅과 릴리스 수명주기
+
+훅을 사용하면 차트 개발자는 릴리스 수명주기 상의 중요 시점에서 
+작업이 수행되도록 할 수 있다. 예를 들어 `helm install` 의 수명주기를 
+생각해보자. 기본적으로 수명주기는 다음과 같다.
+
+1. 사용자가 `helm install foo` 를 실행한다.
+2. 헬름 라이브러리 설치 API가 호출된다.
+3. 검증 후, 라이브러리는 `foo` 템플릿을 렌더링한다.
+4. 라이브러리는 결과로 나온 리소스를 쿠버네티스에 로드한다.
+5. 라이브러리는 릴리스 객체(및 다른 데이터)를 클라이언트에 반환한다.
+6. 클라이언트가 종료된다.
+
+헬름은 `install` 수명주기에 대해 `pre-install` 및 `post-install` 
+의 두 가지 훅을 정의한다. `foo` 차트의 개발자가 두 훅을 모두 구현하면 수명주기는 
+다음과 같이 바뀐다.
+
+1. 사용자가 `helm install foo` 를 실행한다.
+2. 헬름 라이브러리 설치 API가 호출된다.
+3. `crds/` 디렉터리의 CRD가 설치된다.
+4. 검증 후 라이브러리는 `foo` 템플릿을 렌더링한다.
+5. 라이브러리는 `pre-install` 훅 실행을 준비한다. (쿠버네티스에
+   훅 리소스 로딩)
+6. 라이브러리는 가중치(기본값으로는 0을 할당), 리소스 종류, 
+   이름을 기준으로 훅을 오름차순으로 정렬한다.
+7. 라이브러리는 가장 낮은 가중치의 훅을 먼저 로드한다.(음수에서 
+   양수로)
+8. 라이브러리는 훅이 "Ready" 될 때까지 대기한다(CRD 제외).
+9. 라이브러리는 결과로 나온 리소스를 쿠버네티스에 로드한다. 
+   `--wait` 플래그가 설정된 경우 라이브러리는 모든 리소스가 "Ready" 상태가 될 때까지 대기하고, 
+   준비가 될 때까지 `post-install` 훅을 실행하지 않는다.
+10. 라이브러리는 `post-install` 훅을 실행한다. (훅 리소스 로딩)
+11. 라이브러리는 훅이 "Ready" 될 때까지 기다린다.
+12. 라이브러리는 릴리스 객체(및 다른 데이터)를 클라이언트에 반환한다.
+13. 클라이언트가 종료된다.
+
+훅이 준비될 때까지 기다린다는 것은 무엇을 의미하는가? 이것은 
+훅에 선언된 리소스에 따라 다르다. 리소스가 `Job` 또는 `Pod` 종류인 경우 
+헬름은 성공적으로 실행 완료될 때까지 기다린다. 훅이 실패하면 릴리스는 실패하게 된다. 
+이것은 _블로킹(blocking) 작업_이므로 작업이 실행되는 
+동안 헬름 클라이언트는 일시정지될 것이다.
+
+다른 모든 종류의 경우, 쿠버네티스가 리소스를 로드(추가 또는 업데이트)하였다고 
+표시하는 즉시, 리소스는 "Ready" 로 간주된다.
+훅에 많은 리소스가 선언되면 리소스들은 연속적으로 실행된다.
+훅 가중치(아래 참조)가 있으면, 가중치 순서대로 실행된다.
+헬름 3.2.0 부터 같은 가중치를 가진 훅 리소스들은 훅이 없는 일반 리소스와 
+같은 순서로 설치된다. 그밖의 경우에는 순서가 보장되지 않는다. (헬름 2.3.0 
+이상에서는 알파벳순으로 정렬된다. 그러나 이 동작은 바인딩으로 간주되지 않으며 
+향후 변경될 수 있다.) 여기까지가 훅 가중치 추가시 고려할 사항이며,
+만약 가중치가 중요하지 않다면 `0` 으로 설정하는 것이 
+좋다.
+
+### 해당 릴리스에 의해 관리되지 않는 훅 리소스
+
+훅이 생성하는 리소스는 현재 릴리스의 일부로 추적되거나 
+관리되지 않는다. 헬름은 훅이 준비 상태까지 도달했는지를 확인했다면
+훅 리소스를 그대로 남겨둔다. 향후 헬름 3에 릴리스 삭제시 
+훅 리소스에 대한 가비지 수집 기능이 추가될 수 있으므로, 
+삭제되면 안되는 모든 훅 리소스에 `helm.sh/resoure-policy: keep` 어노테이션을 
```

Comment:
This should be: `helm.sh/resource-policy: keep`

Reply:
I will update docs as your comment.